### ISSUE TITLE
Modules without hover type not accessible from API

### ISSUE THREAD

**Issue by AJohnDoe** [MISP/MISP#3197]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Debian
| PHP version               | 7.0
| MISP version / git hash   | 2.4.89

### Expected behavior
Modules accessible through MISP Site should be query-able through the API

### Actual behavior
Modules with only "expansion" as the module-type cannot be queried through the API, failing with "Module not found or not available".

This can be traced back to
https://github.com/MISP/MISP/blob/5f0b92e799e728f24eeaaa25b47f7a83f2e44566/app/Controller/ModulesController.php#L31
which fails because
https://github.com/MISP/MISP/blob/5f0b92e799e728f24eeaaa25b47f7a83f2e44566/app/Controller/ModulesController.php#L17
is hardcoded to check whether the module has the "hover" type.

The above then calls
https://github.com/MISP/MISP/blob/5f0b92e799e728f24eeaaa25b47f7a83f2e44566/app/Model/Module.php#L121-L139
which returns either a module array or a string. When the module being tested does not have the "hover" type, "The requested module is not available for the requested action." is returned.

https://github.com/MISP/MISP/blob/5f0b92e799e728f24eeaaa25b47f7a83f2e44566/app/Controller/ModulesController.php#L31
then blindly checks whether `"The requested module is not available for the requested action."["name"] == $modname` which somehow evaluates to `"T" == $modname` which of course always fails.

There are two issues here:

1. `ModulesController::queryEnrichment` is hardcoded to test whether the module has a "hover" type
2. `Module::getEnabledModule` returns either an array or string

Interestingly the module can still be queried from the MISP Site...

### Steps to reproduce the behavior

1. Create a module without the hover type
2. Hit it through the MISP API eg `curl --header "Authorization: <apikey>" --header "Accept: application/json" --header "Content-Type: application/json" --data @body.json -X POST http://127.0.0.1/modules/queryEnrichment`

**Comment by iglocska** [comment]
Good point, this is because queryEnrichment is the endpoint that was originally used for the hover feature only - will have a look at coming up with something more API friendly.
